### PR TITLE
SA-3608: PSModule Release GitHub Actions Workflow

### PR DESCRIPTION
**Pull request by gweinjc**
## Issues
* [SA-3608](https://jumpcloud.atlassian.net/browse/SA-3608) - PSModule Release GitHub Actions Workflow

## What does this solve?
With the migration from CircleCI to GitHub Actions, we need to transfer our Release workflow into GHA. This workflow will allow us to release new versions of the PowerShell module provided that certain criteria are met.

### Criteria to allow a PowerShell release:
- Pull Request is merged into `master` branch
- Files that were changed were located in the following paths:
    - `"PowerShell/Deploy/**"`
    - `"PowerShell/JumpCloud Module/**"`
    - `"PowerShell/ModuleChangelog.md"`
- The Pull Request is labeled with the `PowerShell Module` label

### Jobs and their function
- `Check-If-Merged`: Validates that the Pull Request was actually merged into Master. If the Pull Request was just closed, we do not want to start a release off of this action.
- `Filter-Branch`: Validates that the PR has PowerShell Module label
- `Check-PR-Label`: Also validates the PR has PowerShell Module label
- `Setup-Build-Dependancies`: Installs required modules for PowerShell release and caches for use in a later job
- `Manual-Approval-Release`: Using the environment `PublishToPSGallery`, all required reviewers will get a notification to approve the workflow to continue with the deployment, otherwise it will timeout or we can manually cancel the workflow in case of accidental release
- `Draft-GH-Release`: Drafts a PowerShell module release using the GitHub Release API/CLI
- `Deploy-Nupkg`: Performs the deployment to PSGallery
- `Cleanup-Cache`: Cleans up any leftover caches from previous runs, this will eliminate any unnecessary resources being used

## Is there anything particularly tricky?
We cannot really test this workflow without intending to release anything.
We do not have CodeArtifact accounted for in this version of the workflow

## How should this be tested?
N/A

[SA-3607]: https://jumpcloud.atlassian.net/browse/SA-3607?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[SA-3608]: https://jumpcloud.atlassian.net/browse/SA-3608?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/release-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+          $PSDependencies = @{
+              'PowerShellGet'                         = @{Repository = 'PSGallery'; RequiredVersion = '3.0.12-beta' }
+              'PackageManagement'                     = @{Repository = 'PSGallery'; RequiredVersion = '1.4.8.1' }
+              'PSScriptAnalyzer'                      = @{Repository = 'PSGallery'; RequiredVersion = '1.19.1' }
+              'PlatyPS'                               = @{Repository = 'PSGallery'; RequiredVersion = '0.14.2' }
+              'AWS.Tools.Common'                      = @{Repository = 'PSGallery'; RequiredVersion = '4.1.122' }
+              'AWS.Tools.CodeArtifact'                = @{Repository = 'PSGallery'; RequiredVersion = '4.1.122' }
+              'JumpCloud.SDK.V1'                      = @{Repository = 'PSGallery'; RequiredVersion = '0.0.35'}
+              'JumpCloud.SDK.V2'                      = @{Repository = 'PSGallery'; RequiredVersion = '0.0.39'}
+              'JumpCloud.SDK.DirectoryInsights'       = @{Repository = 'PSGallery'; RequiredVersion = '0.0.23'}
```
Comment (jworkmanjc) [inline .github/workflows/release-workflow.yml:77]: Eventually, this doesn't have to be today but I think we can keep the SDK releases on latest. I'm not sure they are even required for our Release Pipeline...

I'm pretty sure all were need here is the AWS tools, nuget & powerShellGet

File: .github/workflows/release-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+          . "./PowerShell/Deploy/BuildNuspecFromPsd1.ps1" -RequiredModulesRepo PSGallery
+      - name: Add nuget sources
+        #env:
+          #AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          #AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          #Set-AWSCredential -AccessKey:($env:AWS_ACCESS_KEY_ID) -SecretKey:($env:AWS_SECRET_ACCESS_KEY)
+          #$CARepoEndpoint = "$(Get-CARepositoryEndpoint -Domain jumpcloud-artifacts -Region us-east-1 -Repository jumpcloud-nuget-modules -Format nuget)v3/index.json"
+          #dotnet nuget add source $CARepoEndpoint --name CodeArtifact --username aws --password (Get-CAAuthorizationToken -Domain:("jumpcloud-artifacts") -Region:("us-east-1")).AuthorizationToken
+          dotnet nuget add source "https://www.powershellgallery.com/api/v2/package" --name PSGallery
```
Comment (jworkmanjc) [inline .github/workflows/release-workflow.yml:144]: Nuget is already installed so nice, no need to re-install 
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md

**Review by jworkmanjc — CHANGES_REQUESTED** [review]
I'm good to merge in w/ removal of extra dependancies. At this stage nothing in our Release CI workflow requires:
SDKs.
PSScriptAnalyzer
PlatPS

File: .github/workflows/release-workflow.yml (hunk 1 above)
Comment (jworkmanjc) [reply .github/workflows/release-workflow.yml:77]: We are going to try the cache on merge

**Review by jworkmanjc — APPROVED** [review]
Approved, we'll attempt a release tomorrow

**Review by kmaranionjc — APPROVED** [review]

**Pull request merged**